pkg/http/listener.go

```diff
@@ -88,6 +89,23 @@ type httpListener struct {
 	errorLogFunc           func(error, string, ...interface{}) // function to be called on errors.
 }
 
+// shouldLogError returns false if error is due to a network timeout or
+// io.EOF and true otherwise
+func shouldLogError(err error) bool {
+	switch err := err.(type) {
```

Comment:
I think this can be done more concisely like so:
```go
netErr, ok := err.(net.Error)
if ok {
        if err.Timeout() {
                return false
        }
}

return err != io.EOF
```